### PR TITLE
Setup docker, upgrade to latest phoenix and elixir

### PR DESCRIPTION
**Pull request by ashkan18**
# Docker Image

we are using https://hub.docker.com/r/marcelocg/phoenix/ for our docker image
# Changes
- Had to upgrade phoenix and elixir
- Modified `dev` config to use our real Kafka brokers.
# How to run?

Make sure your docker machine is running locally, then

``` shell
docker-compose build
```

and then 

``` shell
docker-compose up
```

Once everything is compiled, if you are running with [docker for mac](https://docs.docker.com/engine/installation/mac/)  go to:
[http://localhost](http://localhost)

cc: @joeyAghion @bhoggard

### CODE REVIEW

File: config/dev.exs [hunk 1]
```diff
@@ -39,7 +39,7 @@ config :apr, :basic_auth, [
 config :phoenix, :stacktrace_depth, 20
 
 config :kafka_ex,
-  brokers: [{"192.168.99.100", 9092}],
-  consumer_group: System.get_env("KAFKA_CONSUMER_GROUP") || "kafka_ex",
-  disable_default_worker: true,
+  brokers: [{"ip-10-0-0-49.ec2.internal", 9092}, {"ip-10-0-0-248.ec2.internal", 9092}],
```
Comment (bhoggard) [inline config/dev.exs:42]: Could this be set up using environmental variables somehow?

**Comment by bhoggard** [comment]
Since that's not an official Docker image, I would consider basing this on https://hub.docker.com/_/elixir/ and adding in any steps to get everything installed.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+web:
+  build: .
+  command: mix phoenix.server
+  environment:
+    PORT: 80
+  volumes:
+  - .:/app
+  ports:
+  - "80:80"
```
Comment (bhoggard) [inline docker-compose.yml]: Port 80 means you're going to have problems if you run this at the same time as Gravity in Docker locally.

File: config/dev.exs (hunk 1 above)
Comment (ashkan18) [inline config/dev.exs:42]: We actually had it originally as ENV variables, there was an issue when we were setting up CI where ENV variables had to be setup during compile time, so we switched back to this, but I guess with distillery we should test it again.

File: docker-compose.yml (hunk 2 above)
Comment (ashkan18) [inline docker-compose.yml]: I've always got confused with this part of using docker/container, to me it feels like, when I set up a container I shouldn't think about other independent systems, is this right? Basically APR shouldn't know about Gravity.

Comment (bhoggard) [inline docker-compose.yml]: I just think you are better off if you don't re-use ports among your dev apps. We know gravity runs on 80, so I wouldn't expose 80 as the port for this container, which is what the `ports:` directive does.

File: config/dev.exs [hunk 3]
```diff
@@ -39,7 +39,7 @@ config :apr, :basic_auth, [
...
+  consumer_group: System.get_env("KAFKA_CONSUMER_GROUP") || "kafka_ex_local",
```
Comment (izakp) [inline config/dev.exs]: We might want to also append the USER env var as a default here so we each get our own consumer group in development, i.e. `"kafka_ex_local_#{System.get_env("USER")}"`

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      apt-get -y install sudo
+
+# install Node.js (>= 6.0.0) and NPM in order to satisfy brunch.io dependencies
+# See http://www.phoenixframework.org/docs/installation#section-node-js-5-0-0-
+RUN curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash - && apt-get update && sudo apt-get install -y nodejs
+
+ADD . /app
+WORKDIR /app
+ENV PORT 8081
+CMD mix phoenix.server
```
Comment (izakp) [inline Dockerfile:23]: I take it we need to inject the MIX_ENV variable into the container when we run it, so it uses the correct config?

File: config/dev.exs (hunk 1 above)
Comment (izakp) [reply config/dev.exs:42]: We had problems parsing out a string in the form of "host:port,host:port".  We can always try and leave port hardcoded but pass a string in the form of "host,host"

Comment (ashkan18) [reply config/dev.exs:42]: I'd say lets try this again in a separate PR.

**Comment by bhoggard** [comment]
LGTM

**Comment by izakp** [comment]
Cool. Next step would be Circle CI pushing the image to our ECS registry.

**Pull request merged**